components/status/StatusActions.vue

```diff
@@ -23,14 +26,36 @@ const {
   toggleReblog,
 } = $(useStatusActions(props))
 
+function recordEngagement(engagementAction: String) {
+  const analyticsId = feedName ? `${feedName}.post.${engagementAction}` : `post.${engagementAction}`
+  engagement.record({ ui_identifier: analyticsId, mastodon_status_id: status.id, ...engagementDetails[analyticsId] })
+}
+
 function reply() {
+  recordEngagement('reply')
+
   if (!checkLogin())
     return
   if (details)
     focusEditor()
   else
     navigateToStatus({ status, focusReply: true })
 }
+
+function reblog() {
+  recordEngagement('reblog')
+  toggleReblog()
+}
```

Comment:
Do you know if data/product wants to have events for undoing a reblog, favorite or bookmark? Since these functions are toggles, if you un-reblog something it just sends the same engagement event and I'm curious if anyone has mentioned wanting to have the false toggle cases tracked too

Reply:
Me and Aly talked to Kirill. We are going to use separate `ui_identifiers` and for the "undo" actions, the `engagement_type` will be general